types/range-parser/index.d.ts

```diff
@@ -27,7 +27,7 @@ declare namespace RangeParser {
          */
         combine?: boolean;
     }
-    const enum Result {
+    enum Result {
```

Comment:
the source code doesn't have such property and because of that the type definition will not represent the actual code, we need to use literal types here (take a look at my PR attempt - https://github.com/DefinitelyTyped/DefinitelyTyped/pull/29783/files - which you have referenced)